.github/workflows/validator.yml

```
@@ -0,0 +1,72 @@
+
+name: Dataset Validator
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  Validate:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    #- name: Update software
+    #  run: |
+    # do we want to do this? it's helpful but also slow.
+    #    sudo apt-get update &&
+    #    sudo DEBIAN_FRONTEND=noninteractive apt-get -y dist-upgrade &&
+    #    sudo DEBIAN_FRONTEND=noninteractive apt-get autoremove
+
+    - name: Install git-annex
+      run: |
+        sudo apt-get install -y git-annex
+        git config --global annex.thin true
+
+    - name: Install bids-validator
+      run: |
+        # install proper NodeJS for bids-validator
+        curl -sL https://deb.nodesource.com/setup_current.x | sudo -E bash -
+        sudo apt-get install -y nodejs
+        sudo npm install -g bids-validator
+
+    - name: Install spine-generic for acquisition parameters check
+      run: pip install spinegeneric@git+https://github.com/spine-generic/spine-generic.git@master
+
+    #- name: Increase free space
+    #   run: |
+    #    # this takes about 2 minutes but saves about 30GB, which is space we might need for git-annex.
+    #    # annex.thin saves a lot of space, but if the dataset grows beyond what Github can handle
+    #    # try enabling this.
+    #    sudo rm -rf /usr/local/lib/android /usr/share/dotnet
+
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: git config
+      run: |
+        # this is needed by git-annex so that it can write to the local git-annex branch
+        # the tracking information about the local copy
+        git config --global user.name "gh-actions"
+        git config --global user.email "gh-actions"
+        # but actually, disable those writes. We don't need them because we're throwing away this copy immediately.
+        git config annex.alwayscommit false
```

If we're disabling writes, then are the `git config` commands even necessary?

---

The first two? I don't..think...so. I wasn't sure. It seemed safer to leave both in.

---

That's fair! There's not much harm in leaving them in. Was just curious, because I noticed that the old config didn't have `git config annex.alwayscommit false`, so I figured that it was a replacement for the first two lines.

---

You're right. It was.

But I'm not comfortable enough with git-annex to trust that will be enough ;)

---

We cooooould unset them then find out if it breaks (but that would take up more maintenance time and... eh. I agree with leaving them in.)

```suggestion
        # this is needed by git-annex so that it can write to the local git-annex branch
        # the tracking information about the local copy
        git config --global user.name "gh-actions"
        git config --global user.email "gh-actions"
        # but actually, disable those writes. We don't need them because we're throwing away this copy immediately.
        # this might make the previous two lines unnecessary, but it's safer just to leave them both in
        git config annex.alwayscommit false
```